Source/JavaScriptCore/yarr/YarrParser.h

```diff
@@ -584,7 +584,7 @@ class Parser {
          * mode we will allow a hypen to be treated as indicating a range (i.e. /[a-z]/
          * is different to /[a\-z]/).
          */
-        void atomPatternCharacter(char32_t ch)
+        void atomPatternCharacter(char32_t ch, bool hyphenIsRange = false)
```

Comment:
I don't want to have default parameter here. Can you specify this explicitly in the callsites?

Reply:
Thank you. However, in other delegates like [`CharacterClassParserDelegate`](https://github.com/WebKit/WebKit/blob/e1680cddb92b7efc3da3836282ef9d378d400c28/Source/JavaScriptCore/yarr/YarrParser.h#L251), the second argument of `atomPatternCharacter` is optional. To maintain a consistent interface, we need it to be the same in all delegates. Otherwise, a function like [`parserEscape`](https://github.com/WebKit/WebKit/blob/e1680cddb92b7efc3da3836282ef9d378d400c28/Source/JavaScriptCore/yarr/YarrParser.h#L905)  that takes delegate as a template parameter will fail to compile.

Should we make the `atomPatternCharacter` argument in other delegates like `CharacterClassParserDelegate` non-optional as well?

Reply:
Yeah, I think that's better.